github-runner-provisioner/macm1.go

```diff
@@ -42,13 +44,18 @@ set -ex
 
 cd /Users/ec2-user
 
+brew install coreutils
+
 # Download the latest runner package
 curl -o github_runner_installer.tar.gz -L '%[1]s'
 
 # Extract the installer
 sudo su ec2-user -c "tar xzf ./github_runner_installer.tar.gz"
-sudo su ec2-user -c "./config.sh --url https://github.com/%[2]s/%[3]s --token %[4]s --unattended --ephemeral --labels %[5]s" 
-sudo su ec2-user -c "./run.sh"
+sudo su ec2-user -c "./config.sh --url https://github.com/%[2]s/%[3]s --token %[4]s --unattended --labels %[5]s" 
```

Comment:
It won't!

I don't think the runner can do that since the registration token will have expired by that time. I will implement a clean-up mechanism that runs periodically and re-submit this PR.

Reply:
Do tokens time out for removal? The command `./config.sh remove --token %[4]` should be able to clean itself up if run before termination. Do we have any long-lived instance we could try it on?

Reply:
@AidanWenzel I'm testing that on a runner. I'll let it run until tomorrow and see if it removes itself.
Based on some initial tests, the token does not expire when unregistering an agent.

I'd like to get your feedback now. If `./config.sh remove --token` doesn't work, we'll have t look at implementing this on the provisioner, which I'll do in a new PR.